crates/proof-of-sql/src/base/database/table.rs

```diff
@@ -38,15 +45,43 @@ impl<'a, S: Scalar> Table<'a, S> {
     ) -> Result<Self, TableError> {
         Self::try_new(IndexMap::from_iter(iter))
     }
+    /// Creates a new [`Table`] from a [`DataAccessor`], [`TableRef`] and [`ColumnRef`]s.
+    ///
+    /// Columns are retrieved from the [`DataAccessor`] using the provided [`ColumnRef`]s.
+    /// # Panics
+    /// Missing columns or column length mismatches can occur if the accessor doesn't
+    /// contain the necessary columns. In practice, this should not happen.
+    pub(crate) fn from_columns(
+        column_refs: &IndexSet<ColumnRef>,
+        table_ref: TableRef,
+        accessor: &'a dyn DataAccessor<S>,
+        alloc: &'a Bump,
+    ) -> Self {
+        if column_refs.is_empty() {
+            // TODO: Currently we have to have non-empty column references to have a non-empty table
+            // to evaluate `ProofExpr`s on. Once we restrict [`DataAccessor`] to [`TableExec`]
+            // and use input `DynProofPlan`s we should no longer need this.
+            let input_length = accessor.get_length(table_ref);
+            let bogus_vec = vec![true; input_length];
+            let bogus_col = Column::Boolean(alloc.alloc_slice_copy(&bogus_vec));
+            Table::<'a, S>::try_from_iter(core::iter::once(("bogus".parse().unwrap(), bogus_col)))
```

Comment:
Isn't the proper behavior to create a `Table` with no columns, but the proper length?